`no-undef-properties` False positive where `defineModel` is used

**Tell us about your environment**

- **ESLint version:** 8.56.0
- **eslint-plugin-vue version:** 9.20.0
- **Node version:** 18.18.0

**Please show your full configuration:**

```js
module.exports = {
  extends: ['plugin:vue/base'],

  rules: {
    'vue/no-undef-properties': 'error',
  },
};
```

**What did you do?**	

```vue
<template>
  <div id="app">Woof: {{ woof }}</div>
</template>

<script setup>
const model = defineModel();

const woof = computed(() => model.value);
</script>
```

**What did you expect to happen?**

`vue/no-undef-properties` should not throw error.

**What actually happened?**

```
/home/projects/stackblitz-starters-u1b2em/test.vue
  8:35  error  'value' is not defined  vue/no-undef-properties

✖ 1 problem (1 error, 0 warnings)
```

**Repository to reproduce this issue**

https://stackblitz.com/edit/stackblitz-starters-u1b2em

--- a/lib/rules/no-undef-properties.js
+++ b/lib/rules/no-undef-properties.js
@@ -322,25 +322,12 @@ module.exports = {
             propertyReferenceExtractor.extractFromPattern(pattern)
           ctx.verifyReferences(propertyReferences)
         },
-        onDefineModelEnter(node, model) {
+        onDefineModelEnter(_node, model) {
           const ctx = getVueComponentContext(programNode)
 
           ctx.defineProperties.set(model.name.modelName, {
             isProps: true
           })
-
-          if (
-            !node.parent ||
-            node.parent.type !== 'VariableDeclarator' ||
-            node.parent.init !== node
-          ) {
-            return
-          }
-
-          const pattern = node.parent.id
-          const propertyReferences =
-            propertyReferenceExtractor.extractFromPattern(pattern)
-          ctx.verifyReferences(propertyReferences)
         }
       }),
       utils.defineVueVisitor(context, {

--- a/tests/lib/rules/no-undef-properties.js
+++ b/tests/lib/rules/no-undef-properties.js
@@ -558,7 +558,16 @@ tester.run('no-undef-properties', rule, {
       parserOptions: {
         parser: require.resolve('@typescript-eslint/parser')
       }
-    }
+    },
+
+    `
+    <script setup>
+    const model = defineModel();
+    const woof = computed(() => model.value);
+    </script>
+    <template>
+      <div id="app">Woof: {{ woof }}</div>
+    </template>`
   ],
 
   invalid: [

